src/vm/gccover.cpp

```diff
@@ -1175,33 +1262,30 @@ BOOL OnGcCoverageInterrupt(PCONTEXT regs)
     if (gcCover == 0)
         return(FALSE);        // we aren't doing code gcCoverage on this function
 
-    /****
-    if (gcCover->curInstr != 0)
-        *gcCover->curInstr = INTERRUPT_INSTR;
-    ****/
+    BYTE * savedInstrPtr = &gcCover->savedCode[offset];
+
+    // If this trap instruction is taken in place of CORINFO_HELP_STOP_FOR_GC()
+    // Do not start a GC, but continue with the original instruction.
+    // See the comments above SprinkleBreakpoints() function.
+    SLOT nextInstr;
+    SLOT target = getTargetOfCall(savedInstrPtr, regs, &nextInstr);
+
+    if (target == (SLOT)JIT_RareDisableHelper) {
```

Comment:
Instead of doing all this call target decoding, would it be better to just check whether we are in preemptive mode?

The call target decoding will need ongoing tweaks, e.g. I hope to add support for interop transitions for ReadyToRun (based on `CORJIT_FLG2_USE_PINVOKE_HELPERS`) - that will use a different helper.